docs/docs/guides/web3_providers_guide/index.md

```diff
@@ -29,64 +18,136 @@ const web3 = new Web3(/* PROVIDER*/);
+HTTP and WebSocket providers can be supplied as URL strings. All provider types can be supplied by constructing one of the [`SupportedProviders`](/api/web3/namespace/types#SupportedProviders) types.
 
-:::tip
-The passed provider can be either type `string` or one of the [`SupportedProviders`](/api/web3-core#SupportedProviders). And if it is passed as a string, then internally the compatible provider object will be created and used.
-:::
+Keep reading to learn more about the different types of providers and how to use them.
 
 ### HTTP Provider
 
-``` ts title='Initialize HTTP Provider'
+HTTP is a request-response protocol and does not support persistent connection, which means that HTTP providers are not suitable for use cases that require real-time [event subscriptions](/guides/events_subscriptions/).
+
+``` ts title='Initialize an HTTP Provider'
 import { Web3, HttpProvider } from 'web3';
 
+//supply an HTTP provider as a URL string
 //highlight-next-line
 const web3 = new Web3('https://mainnet.infura.io/v3/YOUR_INFURA_ID');
 
 await web3.eth.getBlockNumber()
 // ↳ 18849658n
 
-// or
+// OR
 
+//supply an HTTP provider by constructing a new HttpProvider
 //highlight-next-line
 const web3_2 = new Web3(new HttpProvider('https://mainnet.infura.io/v3/YOUR_INFURA_ID'));
 
 await web3.eth.getBlockNumber()
 // ↳ 18849658n
 ```
 
+#### Configuring HTTP Providers
+
+HTTP providers can be configured by including an [`HttpProviderOptions`](/api/web3-providers-http/interface/HttpProviderOptions/) object in the [`HttpProvider` constructor](/api/web3-providers-http/class/HttpProvider#constructor). The `HttpProviderOptions` type has a single property, `providerOptions`, which is a standard TypeScript [`RequestInit`](https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.requestinit.html) object.
+
+```ts title='Configuring an HTTP Provider'
+import { Web3, HttpProvider } from 'web3';
+
+const httpOptions = {
+    providerOptions: {
+        body: undefined,
+        cache: 'force-cache',
+        credentials: 'same-origin',
+        headers: {
+             'Content-Type': 'application/json',
+        },
+        integrity: undefined,
+        keepalive: true,
+        method: 'GET',
+        mode: 'same-origin',
+        redirect: 'error',
+        referrer: undefined,
+        referrerPolicy: 'same-origin',
+        signal: undefined,
+        window: undefined,
+    } as RequestInit,
+};
+
+const web3 = new Web3(new HttpProvider('https://eth.llamarpc.com', httpOptions));
+```
+
 ### WebSocket provider
 
+WebSockets support a persistent connection between a client and a server, which means they are suitable for use cases that require real-time [event subscriptions](/guides/events_subscriptions/).
+
 ``` ts title='Initialize WS Provider'
 import { Web3, WebSocketProvider } from 'web3';
 
+//supply a WebSocket provider as a URL string
```

Comment:
```suggestion
// supply a WebSocket provider as a URL string
```
nitpick: comments should always have a space before the text

Reply:
Yes, I completely agree 💯 That doesn't seem to be the case in most of the existing docs, though, so I chose to stay consistent. My next PR will be various formatting nitpicks like that 👍🏻